src/orders/components/OrderPayment/components/PaymentsSummary.tsx

```diff
@@ -0,0 +1,87 @@
+import { CardContent } from "@material-ui/core";
+import { OrderDetailsFragment } from "@saleor/graphql";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import SummaryLine from "../../OrderSummaryCard/SummaryLine";
+import { SummaryList } from "../../OrderSummaryCard/SummaryList";
+import { orderPaymentMessages } from "../messages";
+import { useStyles } from "../styles";
+
+interface PaymentsSummaryProps {
+  order: OrderDetailsFragment;
+}
+
+export const getShouldDisplayAmounts = (order: OrderDetailsFragment) => {
```

Comment:
idea: how about returning here also a discriminating union where the discriminant property is the name of the state? We can distinguish here a few states (ordering from the top, example names):
- WRONG_ORDER
- DIFFERENT_AMOUNTS
- NOT_AUTHORIZED
- PARTIAL_CAPTURED
- FULLY_PAID

then in the if statements, you can explicitly say in what state you return null (PaymentsSummary component). 

_happy type safe!_

Reply:
that's really nice, much more readable than just comments in this function